[analyzer][NFC] Modernize loops in LiveVariables analysis

steakhal opened the pull request.

Xazax-hun commented on clang/lib/Analysis/LiveVariables.cpp:
```diff
@@ -90,8 +90,8 @@ namespace {
 
-    for (typename SET::iterator it = B.begin(), ei = B.end(); it != ei; ++it) {
-      A = A.add(*it);
+    for (const auto &element : B) {
```
We have pointers in these sets. Maybe iterating by value would be better to avoid double indirection?

steakhal committed:
[analyzer][NFC] Modernize iterator-based loop in LiveVariables::computeLiveness

steakhal committed:
[analyzer][NFC] Modernize iterator-based loops in LiveVariablesImpl::dumpBlockLiveness

Convert multiple iterator-based loops to range-based for loops:
- DenseMap iteration over blocksEndToLiveness
- std::vector iteration over CFGBlocks
- ImmutableSet iteration over liveDecls
- std::vector iteration over declVec

steakhal committed:
[analyzer][NFC] Modernize iterator-based loop in mergeSets template function

steakhal committed:
Avoid useless references
